### ISSUE TITLE
Pasajeros premium que se encuentran en la fila?

### ISSUE THREAD

**Issue by dchahuan** [IIC2133-PUC/2021-2#33]
En el enunciado dice que los pasajeros premiun entran al tren apenas llegan al anden. Pero si el tren sale se suben solo las personas con tipo normal. Entonces mi pregunta es que pasa con los pasajeros premium que ya se encuentran en el tren.

**Comment by nfragav** [comment]
Supongo que te refieres a los premium que están en el andén, pues los que se encuentran dentro del tren están en sus asientos y no tienen razones para moverse hasta llegar a sus destinos. (nadie les quitará sus asientos)

Lo que pasa con los pasajeros premium es que **nunca** esperan a que parta el tren para subirse a él, por lo que siempre que llegan al andén, entrarán en orden de llegada al tren que se encuentra en el andén parado. Los pasajeros normales no pueden entrar hasta que el tren vaya a partir, por lo que siempre se priorizarán a los premium. Entonces, si se llenan los asientos, del tren y quedan pasajeros premium por subirse, no se subirá ningún pasajero normal en este tren que está por partir. Una vez que parte el tren y llega el siguiente, todos los pasajeros premium que quepan en el tren se subirán en orden de llegada, y si sobran asientos al momento de que el tren parta entonces los pasajeros normales que puedan subirse (por disponibilidad de asientos) se subirán en el orden en que llegaron al andén.

Me disculpo por todo el texto y espero esto te sea útil.

**Comment by dchahuan** [comment]
Me quedo mas claro. Muchas Gracias

**Comment by fran-gl** [comment]
Hola! Entonces después de un evento NUEVO-TREN los pasajeros premium que están en el andén se suben inmediatamente al tren cierto?

**Comment by Huasito-Appel** [comment]
@fran-gl 
Correcto, los premium nunca esperan en el anden, se suben de forma inmediata.
Exito en la tarea!

**Comment by fran-gl** [comment]
Oka, muchas gracias!

**Comment by Jamesmoreno21** [comment]
Hola! Me surgió una duda con esto, entiendo que los premium siempre se suben sin esperar, pero puede ocurrir que el tren este lleno y deban esperar igual en la fila?
Me pongo en los casos en que hayan muchos premium en una fila o de que llegue un tren y se bajen pocos o ningun pasajero. 
Gracias de antemano!

**Comment by nfragav** [comment]
En tal caso esos premium no se suben, porque no caben, y esperan al siguiente tren. Si el siguiente tren tiene asientos disponibles, entonces se suben los premium que puedan subirse.
